### PR TITLE
BAH Education and Work Experience 0994

### PR DESCRIPTION
**Pull request by robfelder**
## Description
This PR addresses the need for an education history and work experience page for the 0994 form.

## Testing done
Local manual & Unit

## Screenshots
screen shot 2019-01-25 at 11 49 05 am
screen shot 2019-01-25 at 11 49 25 am
screen shot 2019-01-25 at 11 49 37 am

## Acceptance criteria
- [x] The user is required to answer employment questions
- [x] The content of the Work and Training page matches the screenshot below
- [x] Work and Training page content: 
image

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/edu-benefits/0994/pages/highTechIndustry.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  },
+  currentHighTechnologyEmployment: {
+    'ui:title':
+      'Have you worked in a high-tech industry in the past couple years?',
+    'ui:widget': 'yesNo',
+    'ui:options': {
+      expandUnder: 'currentEmployment',
+      expandUnderCondition: false,
+    },
+    'ui:required': technologyEmploymentRequired,
```
Comment (zurbergram) [inline src/applications/edu-benefits/0994/pages/highTechIndustry.js]: should this be split out into `utils.jsx`?

Comment (zurbergram) [reply src/applications/edu-benefits/0994/pages/highTechIndustry.js]: is this going to be used somewhere else?

**Review by zurbergram — APPROVED** [review]
LGTM, one question

File: src/applications/edu-benefits/0994/pages/highTechIndustry.js (hunk 1 above)
Comment (robfelder) [reply src/applications/edu-benefits/0994/pages/highTechIndustry.js]: It doesn't have to be. I can move it into here.

**Comment by robfelder** [comment]
@department-of-veterans-affairs/frontend-review-group could I get a code review on this at your earliest convenience. This is dependent on schema changes that can be found [on this branch](https://github.com/department-of-veterans-affairs/vets-json-schema/tree/bah-vettec-edu-work)

Thank you!

**Review by zurbergram — CHANGES_REQUESTED** [review]
The unit tests are breaking because schema for `highTechnologyEmploymentType` does not yet exist in vets-json-schema.  

Please add schema into `vets-website/src/applications/edu-benefits/0994/pages/highTechIndustry.js` for `highTechnologyEmploymentType` and when https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/16484 is being worked there will be coordinated PRs between vets-website and vets-json-schema to make these kind of changes.

**Review by zurbergram — APPROVED** [review]
LGTM

File: src/applications/edu-benefits/0994/pages/highTechIndustry.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      expandUnder: 'currentEmployment',
+      expandUnderCondition: false,
+    },
+    'ui:required': formData => !_.get(formData, 'currentEmployment', false),
+  },
+  'view:salaryEmploymentTypes': {
+    'ui:options': {
+      hideIf: formData => {
+        const showIf =
+          _.get(formData, 'currentEmployment', false) ||
```
Comment (rianfowler) [inline src/applications/edu-benefits/0994/pages/highTechIndustry.js]: - can get rid of this block and return the expression directly e.g. `formData => !(_.get ...`
- should be refactored to fit the semantics of `hideIf`

**Review by rianfowler — CHANGES_REQUESTED** [review]

**Review by rianfowler — APPROVED** [review]
LGTM one minor nit-

File: src/applications/edu-benefits/0994/pages/educationCompleted.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    highestLevelofEducation: {
+      type: 'string',
+      enum: [
+        'High school diploma or GED',
+        'Some college',
+        'Associate’s degree',
+        'Bachelor’s degree',
+        'Master’s degree',
+        'Doctoral degree',
+        'Other',
```
Comment (dmccurdybah) [inline src/applications/edu-benefits/0994/pages/educationCompleted.js]: I think we need to make the enum values more enum-like keys (like `high_school_or_ged`, `some_college`, etc.) to fit how enums are usually defined in the schemas.  We'll have to map those to language on each side.

**Pull request merged**